File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int greatest = 0;
        ArrayList<Boolean> ans = new ArrayList<Boolean>();
        for(int i = 0; i < candies.length; i++){
            greatest = Math.max(candies[i], greatest);
        }
        for(int i = 0; i < candies.length; i++){
            int sum = candies[i] + extraCandies;
            if(sum >= greatest){
                ans.add(true);
            } else{
                ans.add(false);
            }
        }
        return ans;
    }
}

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    let result = [];
    let largest = -1;
    for(let i = 0; i < candies.length; i++){
        if(largest < candies[i]) largest = candies[i];
    }
    for(let i = 0; i < candies.length; i++){
        let newSum = candies[i] + extraCandies;
        if(newSum >= largest) result.push(true);
        else result.push(false);
    }
    return result;
};